src/main/java/com/salesforce/cdp/queryservice/util/HttpHelper.java

```diff
@@ -32,12 +33,14 @@
 @Slf4j
 public class HttpHelper {
 
+    private static final ObjectReader mapperReader = new ObjectMapper().reader();
```

Comment:
will this create problem if there are concurrent executions or 2 parallel executions?

Reply:
I think we should not do this.

Reply:
With this change, Tableau fails with NoSuchMethod error

2021-09-20 18:53:17.513 +0530 (,,,,2,1) grpc-default-executor-1 : ERROR com.tableau.connect.grpc.GrpcProtocolService - Failed in constructProtocol.
java.lang.NoSuchMethodError: 'java.lang.Object com.fasterxml.jackson.databind.ObjectReader.readValue(java.lang.String, java.lang.Class)'
	at com.salesforce.cdp.queryservice.util.HttpHelper.handleSuccessResponse(HttpHelper.java:61) ~[?:?]
	at com.salesforce.cdp.queryservice.util.HttpHelper.handleSuccessResponse(HttpHelper.java:57) ~[?:?]
	at com.salesforce.cdp.queryservice.util.TokenHelper.exchangeToken(TokenHelper.java:175) ~[?:?]
	at com.salesforce.cdp.queryservice.util.TokenHelper.getToken(TokenHelper.java:75) ~[?:?]
	at com.salesforce.cdp.queryservice.util.QueryExecutor.lambda$getTokenWithTenantUrl$0(QueryExecutor.java:175) ~[?:?]
	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:46) ~[?:?]
	at net.jodah.failsafe.RetryPolicyExecutor.lambda$supply$0(RetryPolicyExecutor.java:65) ~[?:?]
	at net.jodah.failsafe.Execution.executeSync(Execution.java:128) ~[?:?]
	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:378) ~[?:?]
	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:68) ~[?:?]
	at com.salesforce.cdp.queryservice.util.QueryExecutor.getTokenWithTenantUrl(QueryExecutor.java:174) ~[?:?]
	at com.salesforce.cdp.queryservice.util.QueryExecutor.executeQuery(QueryExecutor.java:87) ~[?:?]
	at com.salesforce.cdp.queryservice.core.QueryServiceAbstractStatement.executeQuery(QueryServiceAbstractStatement.java:70) ~[?:?]
	at com.salesforce.cdp.queryservice.core.QueryServicePreparedStatement.execute(QueryServicePreparedStatement.java:158) ~[?:?]
	at com.salesforce.cdp.queryservice.core.QueryServiceConnection.isValid(QueryServiceConnection.java:339) ~[?:?]
	at com.salesforce.cdp.queryservice.core.QueryServiceConnection.<init>(QueryServiceConnection.java:58) ~[?:?]
	at com.salesforce.cdp.queryservice.QueryServiceDriver.connect(QueryServiceDriver.java:65) ~[?:?]
	at com.tableausoftware.jdbc.JDBCDriverManager.getConnection(JDBCDriverManager.java:254) ~[jdbcserver.jar:20212.0.8]
	at com.tableausoftware.jdbc.JDBCProtocolImpl.getConnection(JDBCProtocolImpl.java:320) ~[jdbcserver.jar:20212.0.8]
	at com.tableausoftware.jdbc.JDBCProtocolImpl.<init>(JDBCProtocolImpl.java:117) ~[jdbcserver.jar:20212.0.8]